### ISSUE TITLE
Timed traffic lights - warn if segments to short

### ISSUE THREAD

**Issue by originalfoo** [CitiesSkylinesMods/TMPE#302]
When there are very short road segments near timed traffic lights, it seems to break the traffic lights.

It would be good if we could warn the user about problematic segments.

I do not know what the segment length threshold should be, but it's possible to reproduce the error in the save supplied in #297 

I suspect the segment length issue would also impact things like junction restrictions, but have not confirmed this.

**Comment by aubergine10** [comment]
Tagging #8

**Comment by FireController1847** [comment]
Maybe not entirely only warn, but if there's any way we could add support for shorter segments would probably be a much more superior fix than just warning them. This could allow things like cool interchanges with slip-on and slip-off lanes (like the single-point urban interchange) to work. I will try and remember to provide images later.

**Comment by aubergine10** [comment]
Agreed, if we could find what's causing the issue and fix it that would be best.

If not, we should find a way to warn users about the issue.

Images of practical use of short segments will be useful to show why a fix is preferable.

**Comment by Shadertheboi** [comment]
Thank you for the suggestion! For the roads in my country, slip roads are littered at almost every traffic junction:

A typical traffic junction with slip roads for the expressway:
https://goo.gl/maps/C4X2wgkMoGHsxFDq5

A typical traffic junction with slips roads:
https://goo.gl/maps/YvbEMjugwLs4yotk8

If you noticed, this is the exact same junction that I was trying to mimic but failed to do so due to this bug/error where segments are too close to each other.

**Comment by s2500111** [comment]
It would be even better, if there is a possibility to fix the broken behaviour, This error is very frustrating and even happening on segments long enough for like two trucks with trailers.

**Comment by aubergine10** [comment]
There is a possible fix mentioned in #666 - reduce speed on the short segments.

**Comment by s2500111** [comment]
Yeah, found that too, can confirm that lowering the speed limit is working.

**Comment by aubergine10** [comment]
The [Broken Nodes Detector](https://steamcommunity.com/sharedfiles/filedetails/?id=1777173984) mod has a tool which helps users find segments which are too short.

Maybe we could take that code, and when user selects a node to edit TTL, we can highlight any 'too short' segments attached to the node? Or prevent TTL being used on nodes that have a 'too short' segment?

**Comment by krzychu124** [comment]
Fixed in #1730
